src/ucp/wireup/select.c

```diff
@@ -1460,12 +1460,14 @@ ucp_wireup_construct_lanes(const ucp_wireup_select_params_t *select_params,
         addr_indices[lane]            = select_ctx->lane_descs[lane].addr_index;
 
         if (select_ctx->lane_descs[lane].usage & UCP_WIREUP_LANE_USAGE_CM) {
-            if (select_params->ep_init_flags & UCP_EP_INIT_CM_WIREUP_CLIENT) {
-                /* Client creates CM lane before construction TL lanes */
-                ucs_assert((lane == 0) && (key->cm_lane == 0));
-            } else {
-                ucs_assert(key->cm_lane == UCP_NULL_LANE);
+            if (key->cm_lane == UCP_NULL_LANE) {
                 key->cm_lane = lane;
+            } else {
+                ucs_assert(select_params->ep_init_flags &
```

Comment:
move this assert before line 1463?

Reply:
no, this valid only for this branch

Reply:
but don't we always expect to have UCP_EP_INIT_CM_WIREUP_CLIENT flag when we have UCP_WIREUP_LANE_USAGE_CM?
BTW what happens in server side?

Reply:
> but don't we always expect to have UCP_EP_INIT_CM_WIREUP_CLIENT flag when we have UCP_WIREUP_LANE_USAGE_CM?

we do but client and server have difference

> BTW what happens in server side?

both client and server go thru `if (key->cm_lane == UCP_NULL_LANE) {` branch on first configuration, and only client will be "reconfigured" when it gets real address from server.

Reply:
actually, the case where (key->cm_lane != UCP_NULL_LANE) should never happen, we should not have more than one lane desc with CM usage
need to replace if/else with `ucs_assert(key->cm_lane == UCP_NULL_LANE)`, like we're doing for am_lane and tag_lane a few lines below